securesystemslib/signer/_key.py

```diff
@@ -180,22 +215,99 @@ def from_dict(cls, keyid: str, key_dict: Dict[str, Any]) -> "SSlibKey":
     def to_dict(self) -> Dict[str, Any]:
         return self._to_dict()
 
+    def _from_pem(self) -> "PublicKeyTypes":
+        """Helper to load public key instance from PEM-formatted keyval."""
+        public_bytes = self.keyval["public"].encode("utf-8")
+        return load_pem_public_key(public_bytes)
+
+    @staticmethod
+    def _get_hash_algorithm(name: str) -> "HashAlgorithm":
+        """Helper to return hash algorithm for name."""
+        algorithm: HashAlgorithm
+        if name == "sha224":
+            algorithm = SHA224()
+        if name == "sha256":
+            algorithm = SHA256()
+        if name == "sha384":
+            algorithm = SHA384()
+        if name == "sha512":
+            algorithm = SHA512()
+
+        return algorithm
+
+    @staticmethod
+    def _get_rsa_padding(
+        name: str, hash_algorithm: "HashAlgorithm"
+    ) -> "AsymmetricPadding":
+        """Helper to return rsa signature padding for name."""
+        padding: AsymmetricPadding
+        if name == "pss":
+            padding = PSS(mgf=MGF1(hash_algorithm), salt_length=PSS.AUTO)
+
+        if name == "pkcs1v15":
+            padding = PKCS1v15()
+
+        return padding
+
     def verify_signature(self, signature: Signature, data: bytes) -> None:
         try:
-            if not sslib_keys.verify_signature(
-                self.to_securesystemslib_key(),
-                signature.to_dict(),
-                data,
+            if signature.keyid != self.keyid:
+                raise ValueError(
+                    f"keyid mismatch: 'key id: {self.keyid}"
+                    f" != signature keyid: {signature.keyid}'"
+                )
+            sig = bytes.fromhex(signature.signature)
+
+            if CRYPTO_IMPORT_ERROR:
+                if self.scheme == "ed25519":
+                    public_bytes = bytes.fromhex(self.keyval["public"])
+                    checkvalid(sig, data, public_bytes)
+                    return
+
+                raise exceptions.UnsupportedLibraryError(CRYPTO_IMPORT_ERROR)
```

Comment:
I guess another potential structure would be 

```
 if CRYPTO_IMPORT_ERROR:
    if self.scheme != "ed25519":
        raise exceptions.UnsupportedLibraryError(CRYPTO_IMPORT_ERROR)
    try:
        # call vendored ed25519
    except SignatureMismatch as e:
        # handle
    except Exception as e:
        # handle
else:
    try:
        # the actual switch for cryptography calls goes here
    except InvalidSignature as e:
        # handle
    except Exception as e:
        # handle
```
in pseudocode that feels easier to understand (since the try-block is not shared) but in practice I'm unsure if it helps

Reply:
@jku: is 73f84e2 what you had in mind? I'm not sure if it makes things much easier to read.

Reply:
Okay, I think this is good enough. Thanks for your help, Jussi! Lmk if this still looks good to you.